libc-top-half/musl/include/stdlib.h

```diff
@@ -128,9 +128,7 @@ int rand_r (unsigned *);
 
 #if defined(_XOPEN_SOURCE) || defined(_GNU_SOURCE) \
  || defined(_BSD_SOURCE)
-#ifdef __wasilibc_unmodified_upstream /* WASI has no absolute paths */
 char *realpath (const char *__restrict, char *__restrict);
-#endif
```

Comment:
is this intended?

Reply:
Yes -- `realpath` is one of the functions libcxx's `<filesystem>` relies on, although I just realized I neglected to add `realpath.c` to `LIBC_TOP_HALF_MUSL_SOURCES` in the Makefile, and I'm not sure if that implementation is what we want (vs. a stub), so I'll need to take a closer look.

Reply:
it was intentionally disabled in the header. https://github.com/WebAssembly/wasi-libc/pull/82
if you are going to add a realpath implementation anytime soon, it's great. please go ahead.
otherwise, it's probably better to revert this part for now, until the implementation gets ready.

Reply:
Reverting this part causes the libcxx build to fail, so I don't think that's an option (unless we want to patch libcxx).  I've added a stub in https://github.com/WebAssembly/wasi-libc/pull/473, which I believe is the best we can do right now short of patching libcxx.

Reply:
Update: the stub wasn't needed -- turns out musl's `realpath` implementation works fine.